compiler-core/src/javascript/pattern.rs

```diff
@@ -14,8 +14,8 @@ enum Index<'a> {
     Int(usize),
     String(&'a str),
     ByteAt(usize),
-    IntFromSlice(usize, usize),
-    FloatAt(usize),
+    IntFromSlice(usize, usize, bool, bool),
+    FloatFromSlice(usize, usize, bool),
```

Comment:
Convert these to struct variants `IntFromSlice { a: ..., b: ...}` please as I don't understand what the 4 unnamed fields are here